connection/connection.go

```diff
@@ -5,10 +5,12 @@ type ID string
 
 // Connection is a configuration used by the connector to transport data from source to the EG (target).
 type Connection struct {
-	Space  string           `json:"space"`
-	ID     ID               `json:"connectionId"`
-	Target string           `json:"target"`
-	Source AWSKinesisSource `json:"source"`
+	Space            string            `json:"space"`
+	ID               ID                `json:"connectionId"`
+	Target           string            `json:"target"`
+	Type             string            `json:"type"`
+	AWSKinesisSource *AWSKinesisSource `json:"AWSKinesisSource,omitempty"`
```

Comment:
We should follow convention from EG here https://github.com/serverless/event-gateway/blob/master/docs/api.md#register-function

There should be a single "source" field that depending on "type" have a different structure. Also, as it's PoC I don't think it's a priority right now.

Reply:
Thanks for the tip on this one, was thinking about how we could do this. I like your approach in setting a custon `marshal/unmarshal` based on the type. Will update.